src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -2441,3 +2488,86 @@ def request_task_async_callback(
 
     return {"task_queued": True}
+
+
+@router.post(
+    PRIVACY_REQUEST_BULK_SOFT_DELETE,
+    dependencies=[Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_DELETE])],
+    status_code=HTTP_200_OK,
+    response_model=BulkSoftDeletePrivacyRequests,
+)
+def bulk_soft_delete_privacy_requests(
+    *,
+    db: Session = Depends(deps.get_db),
+    client: ClientDetail = Security(
+        verify_oauth_client,
+        scopes=[PRIVACY_REQUEST_DELETE],
+    ),
+    privacy_requests: ReviewPrivacyRequestIds,
+) -> BulkSoftDeletePrivacyRequests:
+    """
+    Soft delete a list of privacy requests. The requests' deleted_at field will be populated with the current datetime
+    and its deleted_by field will be populated with the user_id of the user who initiated the deletion. Returns an
+    object with the list of successfully deleted privacy requests and the list of failed deletions.
+    """
+    succeeded: List[str] = []
+    failed: List[Dict[str, Any]] = []
+
+    user_id = client.user_id
+    if client.id == CONFIG.security.oauth_root_client_id:
+        user_id = "root"
+
+    for privacy_request_id in privacy_requests.request_ids:
+        privacy_request = PrivacyRequest.get(db, object_id=privacy_request_id)
+
+        if not privacy_request:
+            failed.append(
+                {
+                    "message": f"No privacy request found with id '{privacy_request_id}'",
+                    "data": {"privacy_request_id": privacy_request_id},
+                }
+            )
+            continue
+
+        if privacy_request.deleted_at is not None:
+            failed.append(
+                {
+                    "message": f"Privacy request '{privacy_request_id}' has already been deleted.",
+                    "data": {"privacy_request_id": privacy_request_id},
+                }
+            )
+            continue
+
+        privacy_request.soft_delete(db, user_id)
+        succeeded.append(privacy_request.id)
+
+    return BulkSoftDeletePrivacyRequests(succeeded=succeeded, failed=failed)
+
+
+@router.post(
+    PRIVACY_REQUEST_SOFT_DELETE,
+    dependencies=[Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_DELETE])],
+    status_code=HTTP_200_OK,
+    response_model=None,
+)
+def soft_delete_privacy_request(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    client: ClientDetail = Security(
+        verify_oauth_client,
+        scopes=[PRIVACY_REQUEST_DELETE],
+    ),
+) -> None:
+    """
+    Endpoint for soft deleting a privacy request. The request's deleted_at field will be populated with the current datetime
+    and its deleted_by field will be populated with the user_id of the user who initiated the deletion.
+    """
+    privacy_request: PrivacyRequest = get_privacy_request_or_error(
+        db, privacy_request_id
+    )
+    user_id = client.user_id
+    if client.id == CONFIG.security.oauth_root_client_id:
+        user_id = "root"
```

Comment:
I just saw there is a `root_username` field as well if that is useful

Reply:
hmm yeah I wonder which one is more clear: to store user_id for all users but username for the root user, or just this sort of "fake root user id"

Reply:
I think "root" is clear too, up to you